src/webview/js/js.js

```diff
@@ -908,6 +913,20 @@ documentBody.addEventListener('click', (e: MouseEvent) => {
   }
 });
 
+const revealMutedMessages = (target: Element) => {
+  let messageNode = getMessageNode(target);
+  let first_run = true;
+  while (
+    messageNode
+    && messageNode instanceof Element
+    && (messageNode.classList.contains('message-brief') || first_run)
+  ) {
+    messageNode.setAttribute('data-mute-state', 'shown');
+    messageNode = nextMessage(messageNode);
+    first_run = false;
+  }
```

Comment:
The logic controlling this loop is somewhat messy. For example `nextMessage` should always be returning an `Element` which is a message, or else `null` -- so the `instanceof Element` check and the `classList` check shouldn't be needed. (Unless there's an intended contrast between `message-brief` and `message-full`? If so, that calls for an explanatory comment.)

Reply:
Rewrote this as `do { } while ()`, which is much nicer.

The reason for the `instanceof Element` was because of `getMessageNode`, which returns `Node` rather than `Element`, which I've now fixed.

Not sure what you mean about `message-brief` vs `message-full`, the `classList` check is definitely needed. Hopefully the jsdoc makes it clear what's going on?

Reply:
Definitely helpful, thanks!

I'll pick up this thread with a comment on the new version.